### DISCUSSION TITLE
AWS Lambda Deployment

### DISCUSSION THREAD

**Discussion by petrhrobar** [AnswerDotAI/fasthtml#362] [Show and tell]
Hi there,

Amazing work on Fast HTML! Kudos to the entire team for this game-changing project.

I was wondering if there have been any efforts towards deploying Fast HTML? Specifically, I’m interested in deploying it on AWS Lambda, similar to how FastAPI can be deployed using the Mangum package (example: [YouTube tutorial](https://www.youtube.com/watch?v=RGIM4JfsSk0)).

Looking forward to your thoughts!

**Comment by akclace** [comment]
Not sure about Lambda. I have been working on a project https://github.com/claceio/clace. Clace is an application server which can manage containers. To use Clace, the steps are
1. Install and start Clace 
```
curl -L https://clace.io/install.sh | sh
source $HOME/clhome/bin/clace.env
clace server start &
```
2. Make sure Docker or Podman is running

FastHTML apps can be deployed by running 
```
clace app create --approve --spec python-fasthtml --param APP_MODULE=basic_app:app  https://github.com/AnswerDotAI/fasthtml/examples fasthtmlbasic.localhost:/
clace app create --approve --spec python-fasthtml --param APP_MODULE=basic_ws:app  https://github.com/AnswerDotAI/fasthtml/examples fasthtmlws.localhost:/
```

Clace will run both the apps in two different containers. Clace supports staged blue-green deployment. App containers will be stopped automatically when they are idle. Clace gives a Cloud Run type interface on your hardware, allowing you to run hundreds of apps on one machine.

Docs are at https://clace.io/

**Comment by mbuotidem** [comment]
@petrhrobar Here's an AWS Lambda [example](https://github.com/awslabs/aws-lambda-web-adapter/tree/main/examples/fasthtml-response-streaming) with response streaming and cloudformation. And here is a [hello world](https://github.com/mbuotidem/cle) example with terraform
